docs/Containers/Container-Tasks.md

```diff
@@ -0,0 +1,279 @@
+
+-   **resources**: the resources allocated to the container: cpu and
+    memory_mb set the CPU units and the memory (in MB), respectively,
+    that the container is allocated
+
+-   **size**: an alternative to the resources section, a preset size for
+    the container configured within the UI
+
+-   **system**: specification for the CPU architecture and the operating
+    system to be used by your container (currently linux is the only
+    supported operating system)
+
+Once containers are configured, they must be referenced by a build
+variant. Example:
+
+``` yaml
+buildvariants:
+  - name: container-variant
+    display_name: Container Variant
+    run_on:
+      - example-container
+    tasks:
+      - name: test-graphql
+      - name: test-js
+```
+
+The container name must be put in the **run_on** field, in the same way
+that a distro may alternatively be put there. Unlike the distro model,
+where a primary and secondary distro can be specified in this field
+(hence why the field is a list), only one container may be specified in
+the **run_on** field for a containerized variant. If more than one
+container name is specified, only the first entry will be recognized.
+
+### UI Changes
+
+Once configured properly, a variant with container tasks is ready to
+schedule tasks. Once tasks get created, key differences to look for in
+Spruce are:
+
+#### **Container Project Settings**
+
+A new tab has been added to the project settings page for container
+configurations. Users can create a list of resource configuration
+presets that can be referenced via alias in the size field of their
+container YAML configurations.
+
+![containers.png](../images/container_configuration.png)
+
+Options:
+
+-   Name: The alias for the resource preset. Names must be unique within
+    the list.
+
+-   Memory: The amount of memory (in MiB) to allocate.
+
+-   CPU: The CPU units the container can use. These values are expressed
+    in 'vCPU Units'. 1024 CPU units is the equivalent of 1vCPU.
+
+Users can define as many container configurations as needed, reflecting
+different appropriate resource needs for various tasks.
+
+#### **Task Metadata**
+
+A link to a container task's respective container replaces the typical
+host link.
+
+<img alt="containerized_task_metadata.png" height="400" src="../images/container_metadata.png" width="300"/>
+
+#### **Container Page**
+
+The link in the task metadata sidebar takes you to the container page,
+which details the lifecycle of a container and their tasks. Like the
+host page, event logs exist charting the journey of a container task
+from initialization to termination. Task events such as container
+assignment and status changes are also recorded, as well as the clearing
+of a task from a container once it has run its course.
+
+![container_event_logs.png](../images/container_page.png)
+
+### Other Considerations
+
+### Disk Space
+
+Each instance is provisioned with 200GB of space; however, the actual
```

Comment:
Since this is a long explanation of the "why", I would include a one-sentence tl;dr at the top of this section saying something to the effect of "if the task you want to run needs a lot of disk spce, don't use container tasks yet".